### DISCUSSION TITLE
how to use v4.11.1 or v4.12.0

### DISCUSSION THREAD

**Discussion by begemot63** [antlr/antlr4#4209] [Q&A]
When I add to required library dependency in go.mod:
github.com/antlr/antlr4/runtime/Go/antlr/v4 v4.11.1
I got error when running : go mod tidy:

go: downloading github.com/antlr/antlr4/runtime/Go/antlr/v4 v4.11.1
go.oraclecorp.com/microservice-blockchain-connector/internal/core/contract/chaincode/bcapolicy imports
	github.com/antlr/antlr4/runtime/Go/antlr/v4: reading github.com/antlr/antlr4/runtime/Go/antlr/go.mod at revision runtime/Go/antlr/v4.11.1: unknown revision runtime/Go/antlr/v4.11.1

Although such tag 4.11.1 exists in repository :  https://github.com/antlr/antlr4/tree/v4.11.1/runtime/Go/antlr/v4 dependency cannot be solved.
The same occurs for version 4.12.0
How to use  recent version of github.com/antlr/antlr4/runtime/Go/antlr/v4  ?

If I want to get the latest version :
go get  github.com/antlr/antlr4/runtime/Go/antlr/v4
then it loads different version :
go: added github.com/antlr/antlr4/runtime/Go/antlr/v4 v4.0.0-20230321174746-8dcc6526cfb1
\
How to use v4.11.1 or v4.12.0 ?

**Comment by kaby76** [comment]
When you do `go get github.com/antlr/antlr4/runtime/Go/antlr/v4` it gets the correct version: 4.12.0. I did a clone of the repo and a diff -r between the sources in the runtime/Go/antlr/v4 subdirectory and the cached copy that Go stuffs under $USERPROFILE/go/pkg/....... Yes, the go.mod file has a weird timestamp on the name, but that's what I got too. I never trusted "go get" until I found the sources for the runtime that Go caches and compared it with the sources I grabbed from teh repo by git.

You might want to try it from scratch. Delete the go/pkg directory, and start with a generated parser driver. Look at the templates for the Go driver for grammars-v4 [here](https://github.com/antlr/grammars-v4/tree/master/_scripts/templates/Go) as an example. Or, use [trgen](https://github.com/kaby76/Domemtech.Trash/tree/main/src/trgen) to generate a driver for one of the grammars in [grammars-v4](https://github.com/antlr/grammars-v4) (e.g., `git clone https://github.com/antlr/grammars-v4.git; cd grammars-v4/abb; trgen -t Go; cd Generated-Go; make`. Make sure you are on the current version of Go. Mine is `go version go1.19 windows/amd64`. Do not use Antlr 4.10.x nor 4.11.x for Go--major problems were fixed in 4.12. The `dev` branch just had a major PR fix for Go in the last couple of days--even better, but trgen doesn't use the dev branch, and to use 4.12, you have to build the latest version of the Antlr tool .jar because the templates for generating the Go parser from the .g4 were changed.
